src/Stripe.net/Services/OAuth/OAuthAuthorizeUrlStripeUserOptions.cs

```diff
@@ -0,0 +1,267 @@
+namespace Stripe
+{
+    using Newtonsoft.Json;
+
+    public class OAuthAuthorizeUrlStripeUserOptions : INestedOptions
+    {
+        /// <summary>
+        /// <para>The Kana variation of the address block.</para>
+        /// <para>
+        /// This parameter is only relevant for Japan. You must prefill <see cref="Country"/> with
+        /// <c>JP</c> and <see cref="Zip"/> with a valid Japanese postal code to use this parameter.
+        /// </para>
+        /// </summary>
+        [JsonProperty("block_kana")]
+        public string BlockKana { get; set; }
+
+        /// <summary>
+        /// <para>The Kanji variation of the address block.</para>
+        /// <para>
+        /// This parameter is only relevant for Japan. You must prefill <see cref="Country"/> with
+        /// <c>JP</c> and <see cref="Zip"/> with a valid Japanese postal code to use this parameter.
+        /// </para>
+        /// </summary>
+        [JsonProperty("block_kanji")]
+        public string BlockKanji { get; set; }
+
+        /// <summary>
+        /// <para>The Kana variation of the address building.</para>
+        /// <para>
+        /// This parameter is only relevant for Japan. You must prefill <see cref="Country"/> with
+        /// <c>JP</c> and <see cref="Zip"/> with a valid Japanese postal code to use this parameter.
+        /// </para>
+        /// </summary>
+        [JsonProperty("building_kana")]
+        public string BuildingKana { get; set; }
+
+        /// <summary>
+        /// <para>The Kanji variation of the address building.</para>
+        /// <para>
+        /// This parameter is only relevant for Japan. You must prefill <see cref="Country"/> with
+        /// <c>JP</c> and <see cref="Zip"/> with a valid Japanese postal code to use this parameter.
+        /// </para>
+        /// </summary>
+        [JsonProperty("building_kanji")]
+        public string BuildingKanji { get; set; }
+
+        /// <summary>The legal name of the business, also used for the statement descriptor.</summary>
+        [JsonProperty("business_name")]
+        public string BusinessName { get; set; }
```

Comment:
Hum. I have quite a lot of FUD here. We just changed the shape of the Account API and it should now be `business_profile[name]`. We likely made things backwards compatible but I am not a fan of adding old/deprecated options to the library like this.

Sadly, it seems we forgot to add support for new properties in the OAuth flow with the new shape.

Reply:
It's unlikely we'll change the shape of the OAuth API anytime soon.

Reply:
I disagree. Like we should _at least_ support all the new fields explicitly with the new layout and then have only those in the libs IMO. Have you asked Connect first?

Reply:
Connect has confirmed that there are no plans to change the shape of the OAuth API.